gooddata-java-model/src/main/java/com/gooddata/sdk/model/md/visualization/VisualizationConverter.java

```diff
@@ -293,23 +295,23 @@ private static MeasureItem convertMeasureFilters(final MeasureItem measure) {
         return measure;
     }
 
-
-    private static List<CompatibilityFilter> getCompatibilityFilters(final List<FilterItem> filters) {
-        return (List<CompatibilityFilter>)(List<?>) filters;
+    private static List<CompatibilityFilter> getCompatibilityFilters(final List<ExtendedFilter> filters) {
+        return filters.stream()
+            .map(CompatibilityFilter.class::cast)
+            .collect(toList());
     }
 
-    private static List<FilterItem> removeIrrelevantFilters(final List<FilterItem> filters) {
+    private static <T> List<T> removeIrrelevantFilters(final List<T> filters) {
```

Comment:
Why not to use `ExtendedFilter` instead of `T`?

Reply:
Because there are two usages when one is calling it with FilterItem collection and the another one with ExtendedFilter collection. So this was the solution how to avoid changing class types